### DISCUSSION TITLE
노마드코더 영상보고 리액트 연습하는데 콘솔에 react.development.js:245 Warning: Each child in a list should have a unique "key" prop. 오류가 뜨는데 해결방법 혹시 아시는 분 있으신가요...

### DISCUSSION THREAD

**Discussion by furiouskermit** [LearnReactwith/ReactStudy#28] [Q&A] [answered]
const root = document.getElementById("root");

    const h3 = React.createElement("h3", {
      onMouseEnter: () => console.log("yee"),
    }, "Hello I'm a h3");
    const btn = React.createElement("button", {
      onClick: () => console.log("i'm clicked"),
    }, "Click Me");

    const container = React.createElement("div", null, [h3, btn]);

    ReactDOM.render(container, root); 

영상대로 코드를 쳤는데 콘솔에서  Warning: Each child in a list should have a unique "key" prop.
Check the top-level render call using <div>. 라고 뜨며 onClick이랑 onMouseEnter가 안 먹히는데 혹시 어떻게해야 작동하는지 아시는 분 있으실까요
*참고로 위 코드는 react script 코드만 가져온거고 실제 html 안에는 react 링크 다 연결해서 콘솔로 처음에 링크 연결 잘 됐는지 확인했습니다

**Comment by Randalll09** [comment, marked as answer]
const root = document.getElementById('root');

const h3 = React.createElement(
  'h3',
  {
    onMouseEnter: () => console.log('yee'),
    key: '1',
  },
  "Hello I'm a h3"
);
const btn = React.createElement(
  'button',
  {
    onClick: () => console.log("i'm clicked"),
    key: 2,
  },
  'Click Me'
);

const container = React.createElement('div', null, [h3, btn]);

ReactDOM.render(container, root);


배열내에 있는 요소는 list 취급되어서 style 값 부여해주듯 태그 속성으로 key 값을 부여해줘야 하는 걸로 압니다! 
이런식으로 각요소 안에 key 값을 부여해 주시면 오류가 뜨지 않습니다.
그리고 마우스 엔터와 온클릭 함수는 제 컴퓨터에선 잘 작동하는 군요

**Reply by furiouskermit** [reply]
key값은 그럼 임의로 아무거나 정해주기만 하면 되는건가요?

**Reply by Randalll09** [reply]
key값은 각자고유한값을가져야합니다. 
https://ko.reactjs.org/docs/lists-and-keys.html
여기좀더자세한설명이나와있어요!
